extensions/prototype-content-with-issues-table/app.R

```diff
@@ -0,0 +1,197 @@
+library(shiny)
+library(bslib)
+library(DT)
+library(connectapi)
+library(dplyr)
+library(purrr)
+
+# cache data to disk with a refresh every 8h
+shinyOptions(
+  cache = cachem::cache_disk("./app_cache/cache/", max_age = 60 * 60 * 8)
+)
+
+source("get_usage.R")
+
+# Hacky function to get a list of Content class objects without making a request
+# for each item. These objects differ from the ones created by `content_item()`
+# because they also include the full owner info as returned by `get_content()`.
+as_content_list <- function(content_df, client) {
+  cdf_split <- split(content_df, 1:nrow(content_df))
+  map(cdf_split, function(x) {
+    x <- x[, !(names(x) %in% c("tags"))]
+    x <- as.list(x)
+    Content$new(client, x)
+  })
+}
+
+# grab the username associated with a provided user_guid
+get_user_info <- function(client, user_guid, connectServer) {
+  if (is.na(user_guid)) {
+    # NA user_guid populates when ACL is no login required / public access
+    as.character("Anonymous")
+  } else {
+    user_endpoint <- paste0("v1/users/", user_guid)
+    user <- client$GET(user_endpoint)
+    user$username
+  }
+}
+
+# checks to see if a content item has failed jobs, grabs usage data if it does,
+# then compiles content, job, and usage data together, returning it.
+get_failed_job_data <- function(item, usage, client, connectServer) {
+  failed_jobs <- tryCatch(
+    {
+      get_jobs(item) |> 
+        # filter successful jobs
+        filter(exit_code != 0) |> 
+        # map content job types to something more readable 
+        mutate(tag = case_when(
+               tag == "unknown" ~ "Unknown Job Type",
+               tag == "build_report" ~ "Build Report",
+               tag == "build_site" ~ "Build Site",
+               tag == "build_jupyter" ~ "Build Jupyter Notebook",
+               tag == "packrat_restore" ~ "Packrat Cache Restore",
+               tag == "python_restore" ~ "Python Envrionment Restore",
+               tag == "configure_report" ~ "Parametrized Report Configure",
+               tag == "run_app" ~ "Run R Application",
+               tag == "run_api" ~ "Run Plumber API",
+               tag == "run_tensorflow" ~ "Run Tensorflow API Model",
+               tag == "run_python_api" ~ "Run Python API",
+               tag == "run_dash_app" ~ "Run Dash Application",
+               tag == "run_gradio_app" ~ "Run Gradio Application",
+               tag == "run_streamlit" ~ "Run Streamlit Application",
+               tag == "run_bokeh_app" ~ "Run Bokeh Application",
+               tag == "run_fastapi_app" ~ "Run Python FastAPI",
+               tag == "run_pyshiny_app" ~ "Run PyShiny Application",
+               tag == "render_shiny" ~ "Render Quarto or RMarkdown Interactive Document",
+               tag == "run_voila_app" ~ "Run Voila Application",
+               tag == "run_pyshiny_app" ~ "Run PyShiny Application",
+               tag == "ctrl_extraction" ~ "Extract Parameter Controls From Report",
```

Comment:
I wonder if it would be better to bundle these into "Building", instead of "Build Report", "Build Site", and "Build Jupyter Notebook" and then: "Restoring environment", "Running" are there other meta categories we need?